scripts/queuestat

```diff
@@ -35,9 +35,10 @@ STATUS_INVALID = 'INVALID'
 
 QUEUE_TYPE_MC = 'MC'
 QUEUE_TYPE_UC = 'UC'
+QUEUE_TYPE_ALL = 'ALL'
 SAI_QUEUE_TYPE_MULTICAST = "SAI_QUEUE_TYPE_MULTICAST"
 SAI_QUEUE_TYPE_UNICAST = "SAI_QUEUE_TYPE_UNICAST"
- 
+SAI_QUEUE_TYPE_ALL = "SAI_QUEUE_TYPE_ALL"
```

Comment:
Could you please explain what is "All" type? I didn't see such a type on the counter DB in my test. What is the platform and version are you using? In your example, why does some output have UC and MC, while other has "All" only?

Reply:
See the defination from saiqueue.h:
/**
 * @defgroup SAIQUEUE SAI - QOS Queue specific API definitions.
 *
 * @{
 */

/**
 * @brief Enum defining Queue types.
 */
typedef enum _sai_queue_type_t
{
    /** H/w Queue for all types of traffic */
    SAI_QUEUE_TYPE_ALL = 0x00000000,

    /** H/w Unicast Queue */
    SAI_QUEUE_TYPE_UNICAST = 0x00000001,

    /** H/w Multicast (Broadcast, Unknown unicast, Multicast) Queue */
    SAI_QUEUE_TYPE_MULTICAST = 0x00000002,

    /** Custom range base value */
    SAI_QUEUE_TYPE_CUSTOM_RANGE_BASE = 0x10000000

} sai_queue_type_t;

most hardware platform, eg centec switch chip, can support diffenent queue model
1. unicast and multicast flows use separate queues as we know UC MC queue
2. unicast and multicast flows use same queue, this is SAI_QUEUE_TYPE_ALL
3. ...

on centec chip platform, we can config different queue model through a chip profile

Reply:
Thanks for the info!